### PR TITLE
My Community widget: widget deprecation

### PR DESCRIPTION
**Pull request by ivan-ottinger**
This commit hides the My Community widget from the block inserter and Legacy widget block drop-down menu.

**Before**

  Block inserter  |  Legacy Widget block
:-------------------------:|:-------------------------:
Markup on 2022-01-27 at 10:19:36 | Markup on 2022-01-27 at 10:18:48

**After**

  Block inserter  |  Legacy Widget block
:-------------------------:|:-------------------------:
Markup on 2022-01-27 at 10:25:16 | Markup on 2022-01-27 at 10:27:08

The My Community widget on WPCOM will have a dedicated deprecation patch as the My Community widget in Jetpack and My Community widget on WPCOM are not synced.

#### Changes proposed in this Pull Request:
- Add filter to deprecate the My Community widget

#### Jetpack product discussion
- this PR is part of the Legacy Widgets Migration project: pdf5j4-4C-p2
- more info can be found in pdf5j4-7f-p2

#### Does this pull request change what data or activity we track or use?
No, it does not.

#### Testing instructions:
1. Navigate to Customizer → Widgets
2. Add the Legacy block and take a look at the dropdown menu. The My Community widget should not be available.
3. Try to search for the widget in the block inserter in Customizer as well. It should not come up.

### CODE REVIEW

**Review by yansern — DISMISSED** [review]

**Comment by yansern** [comment]
(oh that said, just need the WPCOM patch to go with this.)

**Review by jeherve — APPROVED** [review]

**Comment by ivan-ottinger** [comment]
> (oh that said, just need the WPCOM patch to go with this.)

Yes, I have it on my todo list. 👍🏼

**Comment by ivan-ottinger** [comment]
Hey @jeherve, I had to do the rebase. If you have a chance, feel free to approve this PR once again - so we can merge it. Thank you. :)

**Review by jeherve — APPROVED** [review]

**Comment by ivan-ottinger** [comment]
Related WPCOM patch: D73981-code.

**Comment by ivan-ottinger** [comment]
Just noting here that I have requested the responsible team (93VbO-p2) to unpublish the related doc page: https://jetpack.com/support/extra-sidebar-widgets/my-community-widget/

**Pull request merged**
